bootstrap.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -euo pipefail
+
+#######################################################################################################################
+# Crate Universe
+#######################################################################################################################
+
+# Find the location of the script
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+BOOTSTRAP_DIR="${SCRIPT_DIR}/crate_universe/private/bootstrap"
+
+# Go to root of rules_rust
+pushd "${SCRIPT_DIR}" &> /dev/null
+has_changes="$(git log origin/main.. -- crate_universe/**)"
+popd &> /dev/null
+
+# Only bootstrap the binary if there are changes
+if [[ -n "${has_changes}" ]]; then
+    pushd "${BOOTSTRAP_DIR}" &> /dev/null
+    bazel run //:build && bazel run //:install && bazel clean
+    popd &> /dev/null
+
+    # Generate a bazelrc file for ensuring we always use the resolver binary for the current host
+    cat << EOF > "${SCRIPT_DIR}/crate_universe.bazelrc"
```

Comment:
I intentionally added it in both places but only because I wasn't sure how this would shake out. We currently run examples from the root of the repo which I feel is not ideal. If we want to continue that then we'd need the `.bazelrc` file in the root.